app/gui2/shared/ast/tree.ts

```diff
@@ -1825,6 +1826,10 @@ export class MutableNumericLiteral extends NumericLiteral implements MutableAst
 export interface MutableNumericLiteral extends NumericLiteral, MutableAst {}
 applyMixins(MutableNumericLiteral, [MutableAst])
 
+export function isNumericLiteral(s: string) {
+  return is_numeric_literal(s.trim())
```

Comment:
Not sure about trim here. Submitting code expresisons with trailing spaces can cause issues, so effectively ignoring them in this method seems bad. If you want to trim the inputted code, do it before calling this method.